### PR TITLE
stage:list: update output and help text

### PR DESCRIPTION
**Pull request by skshetry**
* Replaced 'Produces' with 'Outputs'
* Replaced 'Stage depends on' with just 'Depends on'
* Changes help text of --names-only and --fail

Addressing some of the review comments from https://github.com/iterative/dvc/pull/5282#pullrequestreview-571831191.

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/command/stage.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def is_plot_or_metric(out: "BaseOutput"):
 
     outs = list(filterfalse(is_plot_or_metric, stage.outs))
     if outs:
-        desc.append("Produces " + part_desc(outs))
+        desc.append("Outputs " + part_desc(outs))
```
Comment (skshetry) [inline dvc/command/stage.py:37]: @jorgeorpinel, still not sure about `metrics/params` part, so I am leaving `Generates` part as-is. 

For others, here's the discussion around it: https://github.com/iterative/dvc/pull/5282#discussion_r560641053

Comment (dberenbaum) [reply dvc/command/stage.py:37]: I don't see why "Generates" is any more specific to metrics or plots than "Outputs." Maybe "Reports" would be a bit more specific, but I'm not sure it's worth it to have a different verb. That being said, this is truly nitpicking and seems unlikely to even be noticed by users, so I'm fine to leave as is if this is your preference @skshetry .

File: dvc/command/stage.py [hunk 2]
```diff
@@ -348,7 +348,7 @@ def add_parser(subparsers, parent_parser):
         "--fail",
         action="store_true",
         default=False,
-        help="Fail immediately if there's an error.",
+        help="Do not suppress any syntax errors, fail immediately.",
```
Comment (dberenbaum) [inline dvc/command/stage.py:351]: Again nitpicking, but I'd suggest switching the order so the primary effect of the flag is obvious from the first couple words.

```suggestion
        help="Fail immediately, do not suppress any syntax errors.",
```

**Review by dberenbaum — APPROVED** [review]
Left a couple very minor comments, but up to you to decide whether to accept them @skshetry .

File: dvc/command/stage.py (hunk 1 above)
Comment (skshetry) [reply dvc/command/stage.py:37]: I think `Reports` is what I am looking for, though I find it to be a bit _formal_. Thanks @dberenbaum.

**Pull request merged**